### PR TITLE
Add tests for action_listener.0.dart

### PR DESCRIPTION
**Pull request by ValentinVignal**
Contributes to https://github.com/flutter/flutter/issues/130459

It adds a test for
- `examples/api/lib/widgets/actions/action_listener.0.dart`

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: examples/api/test/widgets/actions/action_listener.0_test.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    await tester.tap(find.text('Disable'));
+    await tester.pump();
+
+    expect(find.widgetWithText(OutlinedButton, 'Enable'), findsOne);
+    expect(
+      log,
+      const <String?>['Action Listener was added', 'Action Listener was removed'],
+    );
+
+      debugPrint = originalDebugPrint;
```
Comment (bleroux) [inline examples/api/test/widgets/actions/action_listener.0_test.dart:47]: To properly restore debugPrint, I think the current solution is to rely on try/finally.
See https://github.com/flutter/flutter/blob/b2b0a751ff70816f793f372b8d2776bc1a3263f3/packages/flutter/test/widgets/focus_manager_test.dart#L2257

Comment (ValentinVignal) [reply examples/api/test/widgets/actions/action_listener.0_test.dart:47]: That's indeed safer ! Thank you ! 

I made the change in [refactor: Use try-finally](https://github.com/flutter/flutter/pull/150606/commits/576696e2e459556f8388c6bc4402ed9aa4996961)

**Review by bleroux — APPROVED** [review]
LGTM!

File: examples/api/test/widgets/actions/action_listener.0_test.dart [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/widgets/actions/action_listener.0.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('It should enable the action listener, trigger it and disable it', (WidgetTester tester) async {
```
Comment (TahaTesser) [inline examples/api/test/widgets/actions/action_listener.0_test.dart:11]: What is "It"?

Comment (ValentinVignal) [reply examples/api/test/widgets/actions/action_listener.0_test.dart:11]: "It" means "the test", "the widgetTester". How would prefer the description to be ?

Comment (TahaTesser) [inline examples/api/test/widgets/actions/action_listener.0_test.dart:11]:
```suggestion
  testWidgets(''ActionListener can be enabled, triggered, and disabled', (WidgetTester tester) async {
```

**Review by TahaTesser — APPROVED** [review]
LGTM with updated title

File: examples/api/test/widgets/actions/action_listener.0_test.dart [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/widgets/actions/action_listener.0.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets(''ActionListener can be enabled, triggered, and disabled', (WidgetTester tester) async {
```
Comment (bleroux) [inline examples/api/test/widgets/actions/action_listener.0_test.dart:11]:
```suggestion
  testWidgets('ActionListener can be enabled, triggered, and disabled', (WidgetTester tester) async {
```

**Pull request merged**